torch/nn/modules/transformer.py

```diff
@@ -171,11 +173,10 @@ def forward(self, src, mask=None, src_key_padding_mask=None):
         """
         output = src
 
-        for i in range(self.num_layers):
-            output = self.layers[i](output, src_mask=mask,
-                                    src_key_padding_mask=src_key_padding_mask)
+        for mod in self.layers:
+            output = mod(output, src_mask=mask, src_key_padding_mask=src_key_padding_mask)
 
-        if self.norm:
+        if self.norm is not None:
```

Comment:
Is this change due to `jitable`?

Reply:
Right the check has to be that the value is not None so this branch doesn't get compiled if there is no code to call (i.e. if `norm` is `None`)